### PR TITLE
fix: Util.parseEmoji

### PR DESCRIPTION
**Pull request by matthewfripp**
**Please describe the changes this PR makes and why it should be merged:**
fixes Util.parseEmoji, currently if you provide a string like "hello this is a ❤ emoji" it would return:
```js
{ animated: false, name: 'hello this is a ❤ emoji', id: null }
```
which is unexpected behaviour, as it should return "❤" as `.name`, this PR fixes that.

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/Util.js [hunk 1]
```diff
@@ -242,11 +242,14 @@ class Util {
    * @private
    */
   static parseEmoji(text) {
-    if (text.includes('%')) text = decodeURIComponent(text);
-    if (!text.includes(':')) return { animated: false, name: text, id: null };
-    const m = text.match(/<?(?:(a):)?(\w{2,32}):(\d{17,19})?>?/);
-    if (!m) return null;
-    return { animated: Boolean(m[1]), name: m[2], id: m[3] || null };
+    if (text !== decodeURIComponent(text)) text = decodeURIComponent(text);
+    // eslint-disable-next-line max-len
```
Comment (DevYukine) [inline src/util/Util.js]: You should not disable max-len and instead split the statement onto 2 lines

File: src/util/Util.js [hunk 2]
```diff
@@ -242,11 +242,14 @@ class Util {
    * @private
    */
   static parseEmoji(text) {
-    if (text.includes('%')) text = decodeURIComponent(text);
-    if (!text.includes(':')) return { animated: false, name: text, id: null };
-    const m = text.match(/<?(?:(a):)?(\w{2,32}):(\d{17,19})?>?/);
-    if (!m) return null;
-    return { animated: Boolean(m[1]), name: m[2], id: m[3] || null };
+    if (text !== decodeURIComponent(text)) text = decodeURIComponent(text);
```
Comment (DevYukine) [inline src/util/Util.js]: use a variable here to store decodeURIComponent instead calling it 2 times

File: src/util/Util.js (hunk 1 above)
Comment (matthewfripp) [reply src/util/Util.js]: I shortened the regex down, it's 9 chars over now. Is disabling still not okay?

**Comment by appellation** [comment]
The current behavior is *undefined* not unexpected. The docs clearly state:

1. This function is private, and the library never needs to parse the emojis out of a particular string.
2. It only accepts 3 different forms of emoji: UTF-8, URL-encoded UTF-8, and custom Discord emojis. None of these include strings that are not wholly a single emoji.

This function was never intended for use with non-emoji strings and I see no reason to change that now. Even if this change is desired, I would still contend that this introduces a lot of ambiguity since it does not handle cases where the string contains multiple emojis.

**Pull request closed without merging**